### DISCUSSION TITLE
Updation specific node feature (multiple node features)

### DISCUSSION THREAD

**Discussion by kimnupjuk** [pyg-team/pytorch_geometric#5023] [Q&A]
Hello, I'm trying to implement 3DGCN (https://arxiv.org/abs/1811.09794) with PyG.

The model deals with two features, scalar feature and vector feature.

The model updates one feature by mixing it with other features (eg. scalar feature is updated with previous scalar feature and vector feature).

So, in the update phase, I think we should assign which feature to be updated with a specific message.

Is there any simple way to do that? 

In Tutorial and other examples, I think there isn't any model to deal with multiple node features.

Or, PyG automatically decides which feature to be updated, because two features(messages) have different shapes.

Thank you!

**Comment by rusty1s** [comment]
PyG can handle multiple features. All features that you pass to propagate are available in message, aggregate and update. Does that help you implement your method?

**Reply by kimnupjuk** [reply]
thank you for your kind reply!
However, my question was how to update specific features.
If I make a message passing layer that makes message for the scalar feature, how can I update scalar feature, not vector feature.
can you show me simple example to assign it?
I think PyG tutorial doesn't deal with this

Thank you!

**Reply by rusty1s** [reply]
I think in this case message would return a tuple of scalar and vector messages. You can then override aggregate (since it now takes in a tuple as input), and apply the aggregation individually. See here for an example: https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_message_passing.py#L429
